druid/src/core.rs

```diff
@@ -908,7 +930,7 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
                     } else {
                         if self.state.children_changed {
                             self.state.children.clear();
-                            self.state.focus_chain.clear();
+                            self.state.reset_focus_chain();
```

Comment:
I changed the behavior again, because of your other reviews. Now the focus-chain collects the focus-chains from its children first and than prepends its own id if auto-focus is set. This works better, than appending the own value in the beginning since in case of `WidetAdded`, the auto_focus flag is is not set, then. Is this ok?

Reply:
> Can you write down a quick summary of what the new expectations are around the focus chain? how does registration work, when does it happen, etc?

Should I write that in the docs?

Reply:
It's a good question where this should be. Maybe in the docs for like `RouteWidgetAdded`? Or otherwise just as a comment in the `WidgetState` struct, where all the fields are defined?

Reply:
Then i will write it in WidgetState